### PR TITLE
Fix near-plane clipping for GLES

### PR DESCRIPTION
**Pull request by Waterdish**
Tiny change, but fixes multiple clipping issues as seen in the links below.

https://github.com/Waterdish/Shipwright-Android/issues/6#issuecomment-1803711439
https://github.com/HarbourMasters/2ship2harkinian/discussions/556

### CODE REVIEW

**Comment by awe444** [comment]
PXL_20240529_145550362

I compiled a test build (GLES on Ubuntu ARM64, Odroid Go Ultra handheld) and the HUD elements now display properly.

**Comment by awe444** [comment]
Picto Box images are still rendered all black, but that's probably a separate issue with a different required fix.

**Review by briaguya0 — APPROVED** [review]

File: src/graphic/Fast3D/gfx_opengl.cpp [hunk 1]
```diff
@@ -342,6 +342,9 @@ static struct ShaderProgram* gfx_opengl_create_and_load_new_shader(uint64_t shad
         vs_len += sprintf(vs_buf + vs_len, "vInput%d = aInput%d;\n", i + 1, i + 1);
     }
     append_line(vs_buf, &vs_len, "gl_Position = aVtxPos;");
+#if defined(USE_OPENGLES) // workaround for no GL_DEPTH_CLAMP
```
Comment (Kenix3) [inline src/graphic/Fast3D/gfx_opengl.cpp:345]: Is this because opengles doesn't support `GL_DEPTH_CLAMP`?

Comment (Waterdish) [reply src/graphic/Fast3D/gfx_opengl.cpp:345]: Yes

**Pull request merged**